File: AddExample1/share/templates/c++/main.cpp
// This is the test program your coded will be run against. If you modify it
// locally just remember your changes won't be included when the grading test is run.
#include <cctype>
#include "vy_tests.hpp"
#include "add.cpp"

// Test program
int main(int argc, char *argv[]) {
  if (argc != 3) {
    std::cout << argv[0] << " requires two inputs: a path to the input file, a path to the output file" << std::endl;
    return 1;
  }
  // Load tests from file
  nlohmann::json data = vy_tests::parse_json_file(argv[1]);
  vy_tests::write_json_file(argv[2], 1, data);

  // Run the problems
  for (auto &problem : data["problems"]) {
    double a = problem["inputs"].value("a",-1.0);
    double b = problem["inputs"].value("b",1.0);
    double c = add(a,b);
    if (problem.find("outputs") != problem.end()) {
      vy_tests::grade_problem(problem["outputs"], "c", c);
    }
  }

  // Build some plots
  vy_tests::write_json_file(argv[2], 1, data);
  return 0;
}

File: AddExample1/share/templates/python/main.py
# I suggest that you don't modify this file! It is used to test your code locally
# and any changes you make won't be used in the actual grading. This file:
#  1. loads some test inputs 
#  2. executes your code on those inputs
#  3. stores the result
import json, sys
from pathlib import Path
import add                  # <--- Loading your function!
import vy_tests             # <---- Loads built-in helper function

if __name__ == '__main__':
  if len(sys.argv) != 3:
    print(sys.argv,"requires two inputs: a path to the input file, a path to the output file")
    sys.exit(1)

  # Read an input json file containing problems
  data = json.loads(Path(sys.argv[1]).read_text())
  Path(sys.argv[2]).write_text(json.dumps(data,indent=2))

  # Run the problems
  for problem in data.get("problems",[]):
    if "inputs" in problem:
      a = problem["inputs"].get("a",-1.0)
      b = problem["inputs"].get("b",1.0)
      c = add.add(a, b)
      if 'outputs' in problem:
        vy_tests.grade_problem(problem["outputs"], "c", c)

  Path(sys.argv[2]).write_text(json.dumps(data,indent=2))
